src/services/workspace-client/devWorkspaceClient.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2018-2020 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { injectable } from 'inversify';
+import { convertDevWorkspaceV2ToV1, isDeleting, isWebTerminal } from '../../devworkspace';
+import { WorkspaceClient } from './';
+import { DevWorkspaceClient as DevWorkspaceClientLibrary } from '@eclipse-che/devworkspace-client';
+import { DevWorkspaceApi } from '@eclipse-che/devworkspace-client/dist/api';
+import { WorkspaceStatus } from '../helpers/types';
+
+interface IStatusUpdate {
+  error?: string;
+  status?: string;
+  prevStatus?: string;
+  workspaceId: string;
+}
+
+/**
+ * This class manages the connection between the frontend and the devworkspace typescript library
+ */
+@injectable()
+export class DevWorkspaceClient extends WorkspaceClient {
+
+  private devworkspaceClient: DevWorkspaceApi;
+  private previousItems: Map<string, Map<string, IStatusUpdate>>;
+
+  constructor() {
+    super();
+    this.devworkspaceClient = DevWorkspaceClientLibrary.getApi(this.axios);
+    this.previousItems = new Map();
+  }
+
+  getAllWorkspaces(defaultNamespace: string): Promise<any> {
+    return this.devworkspaceClient.getAllWorkspaces(defaultNamespace).then((workspaces: any) => {
```

Comment:
Could you specify an interface for workspaces?
For example:
```
 workspaces: che.DevWorkspace[];
 ```

Reply:
Could we use the async function instead of ```.then```?

Reply:
For example:

```
  async getAllWorkspaces(defaultNamespace: string): Promise<che.Workspace[]> {
    const workspaces = await this.devworkspaceClient.getAllWorkspaces(defaultNamespace);

    return workspaces.filter(workspace => !isDeleting(workspace) && !isWebTerminal(workspace)).map(workspace => convertDevWorkspaceV2ToV1(workspace));
  }
  ```
  
 WDYT?

Reply:
I agree with Oleksii on these points: using type `any` leads to decreasing project reliability; using async-await instead of promise chaining makes code more readable.